### ISSUE TITLE
Temporary close permanently stops mysql access

### ISSUE THREAD

**Issue by Lazarus404** [Eonblast/Emysql#164]
Hi,

So, I've been running an app I wrote for some time, and this app breaks via emysql.  After a number of hours of running, I get this in the erlang logs:

```
application: emysql
exited: stopped
type: temporary
```

Once this happens, no more writes to mysql occur and the app essentially breaks.  I have a gen_server which creates the mysql pool in its init function, and looks something like this:

```
init([
    {
        MHostName,
        MPort,
        MUser,
        MPassword,
        MDatabaseName
    } ->
    application:start(emysql),
    emysql:add_pool(auth_pool, 10,
        MUser, MPassword, MHostName, MPort,
        MDatabaseName, utf8).
```

Then, in each handle_info function, I do a simple emysql stored procedure call, like:

```
emysql:execute(auth_pool, erlang:list_to_binary(Query))
```

Is there a way to force the mysql connection to reopen on close or to at least discover if the connection is closed so I can force the gen_server to kill itself abnormally?

Thanks,
Lee

**Comment by jlouis** [comment]
I'd go look at the error log for your system. If an application stops, it is usually some grave error that happens in the application. They are not supposed to stop like that. If your application depends critically on emysql being up, you should perhaps start it as a permanent application:

```
application:start(emysql, permanent)
```

which would have killed your node once emysql fails.

**Comment by Lazarus404** [comment]
Thanks for the response.  I've updated the app so emysql is permanent.  
Hopefully, that'll at least get the thing restarting when it's needed.  
I've checked the logs on the server and there is nothing related to 
emysql.  This is the problem, I think.  The app continues as those 
emysql was still able to connect to the database.  The database is on 
another server, which I don't think is helping, sometimes.

I'll let you know if I'm able to dig up more.

Thanks,
Lee

**Comment by Lazarus404** [comment]
No, it still ended.  3 hours after starting.  It always seems to stop at about the same time after the app starts.  Is there a way to check if the connection is closed via emysql so I can force the gen_server to restart?

Thanks
